src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,52 +40,1012 @@ const ActorInit En_Niw_InitVars = {
+            if (attackNiw) {
+                this->unk290 += 1;
+                this->unkTimer252 = 10;
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_80891974.asm")
+// test
+void func_808917F8(EnNiw* this, GlobalContext* globalCtx, s32 arg2) {
+    f32 phi_f2;
+    f32 targetRotY;
+    f32* D_8089348CPtr = D_80893486;
 
+    if (this->unkTimer250 == 0) {
+        this->unkTimer250 = 3;
+        if ((this->actor.bgCheckFlags & 1)) {
+            // hit floor
+            this->actor.velocity.y = 3.5f; // hop up?
+        }
+    }
+    if (this->unkTimer252 == 0) {
+        this->unk29A++;
+        this->unk29A &= 1;
+        this->unkTimer252 = 5;
+    }
+    if (this->unk29A == 0) {
+        phi_f2 = D_8089348CPtr[arg2];
+    } else {
+        phi_f2 = -D_8089348CPtr[arg2];
+    }
+    if (arg2 == 1) {
+        if ((this->unkTimer254 == 0) || ((this->actor.bgCheckFlags & 8))) {
+            this->unkTimer254 = 150;
+            if (this->yawTimer == 0) {
+                this->yawTimer = 70;
+                this->yawTowardsPlayer = this->actor.yawTowardsPlayer;
+            }
+        }
+    }
+    targetRotY = this->yawTowardsPlayer + phi_f2;
+    Math_SmoothStepToS(&this->actor.world.rot.y, targetRotY, 3, this->unk300, 0);
+    Math_ApproachF(&this->unk300, 3000.0f, 1.0f, 500.0f);
+    func_80891320(this, globalCtx, 5);
+}
+
+void EnNiw_SetupIdle(EnNiw* this) {
+    SkelAnime_ChangeAnim(&this->skelanime, &D_060000E8, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_060000E8.common), 0,
+                         -10.0f);
+    this->unknownState28E = 0;
+    this->actionFunc = func_808919E8;
+}
+
+#ifdef NON_MATCHING
+// non-matching: stack offset and regalloc
+// EnNiw_Idle
+void func_808919E8(EnNiw* this, GlobalContext* globalCtx) {
+    s16 s16tmp;
+    Vec3f newPos;
+
+    newPos.y = randPlusMinusPoint5Scaled(100.0f);
+    newPos.z = randPlusMinusPoint5Scaled(100.0f);
+    if (this->niwType == ENNIW_TYPE_REGULAR) {
+        if (Actor_HasParent(&this->actor, globalCtx)) {
+            // picked up
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M); // crow
+            this->sfxTimer1 = 30;
+            this->unkTimer250 = 30;
+            this->actor.flags &= ~0x1; // targetable OFF
+            this->unknownState28E = 4;
+            this->actor.speedXZ = 0.0f;
+            this->actionFunc = EnNiw_Held;
+            return;
+        } else {
+            func_800B8BB0(&this->actor, globalCtx);
+        }
+    } else {
+        this->unkTimer252 = 10;
+    }
+
+    s16tmp = 0;
+    if (this->unkTimer252 != 0) {
+        if (Rand_ZeroFloat(3.99000000954f) < 1.0f) {
+            this->unk2EA++;
+            this->unk2EA &= 1;
+        }
+        Math_ApproachF(&this->unk264[9], D_80893484[this->unk2EA], 0.5f, 4000.0f);
+    }
+
+    if ((this->unkTimer252 == 0) && (this->unkTimer250 == 0)) {
+        this->unk298 += 1;
+
+        if (this->unk298 >= 8) {
```

Comment:
```suggestion
        if (this->unk298 > 7) {
```
Since `this->unk298` is set to `7` below, I think having `7` here is more relevant